eats_worm/Extractor.py

```diff
@@ -467,9 +473,60 @@ def __init__(self, *args, **kwargs):
             pass
         else:
             kwargs['regen']=True
+            print(os.path.join(self.output_dir, 'params.json'))
             with open(os.path.join(self.output_dir, 'params.json'), 'w') as json_file:
                 json.dump(kwargs, json_file)
 
+    def process_im(self):
+        """Runs basic image pre-processing steps and saves processed image"""
+        if self.processing_params["neuroPAL"]:
+            channels = self.processing_params["RGBW_channels"]
+            NP_image = np.transpose(self.im.tf[0].asarray())
+            print(self.im.tf[0].asarray().shape)
+            NP_image = np.squeeze(NP_image[:,:,:,channels])
+            NP_image = NP_image.astype('int32')
+
+            print('Preprocessing image')
+
+            if self.processing_params["resample"]:   
+                print('Resampling image') 
+                new_res = self.processing_params["new_resolution"]
+                old_res = self.processing_params["old_resolution"]
+                NP_image = res.zoom_interpolate(new_res, old_res, NP_image)
+
+                print('Resampled image to: '+str(new_res))
+
+            if self.processing_params["median_filter"]:
+                print('Median filtering image')
+                size = self.processing_params.get("median", 3)
+
+                NP_image = medFilter2d(NP_image, size)
+
+            if self.processing_params["histogram_match"]:
+                # TODO: make sure reffiles are in tif format
+                print('Matching histogram of image to reference')
+
+                A_max = self.processing_params["A_max"]
+                ref_max = self.processing_params["ref_max"]
+
+                reffile = sio.loadmat(self.processing_params["im_to_match"])
```

Comment:
Eventually we will probably have everything reference NWB files. Currently, we only have the reference neuroPAL files in .mat format which is why it is loadmat here, but you're right that this should probably be more flexible. I might need to think a little more about how to handle this practically in the mean time because I don't think it's a great idea to just have if statements for every possible file format

Reply:
Cool, that makes sense to me. Can you just add a comment to that effect and open an issue in the tracker to track future generalization work? We don't need to get bogged down with future proofing before shipping these changes, but we should leave an explanation in the code base when we make stopgap implementation decisions based on external context